pkg/boot/measurement.go

```diff
@@ -0,0 +1,150 @@
+package boot
```

Comment:
this seems so generally useful (measuringreader) I wonder why it's not in the cpio package.

Reply:
Right now, the measuring reader is specific to the netboot21 format, with a signature and signature_algo file in the cpio. Maybe we can make this more general at some point later.